File: Models/Post.cs
using System;

namespace WebForums.Models
{
	public class Post
	{
		public int ID { get; set; }
		public virtual Thread Thread { get; set; }
		public string Content { get; set; }
		public User Poster { get; set; }
		public DateTime Created { get; set; }
		public DateTime Update { get; set; }
	}
}

File: src/entities/Post.ts
import { Entity, PrimaryGeneratedColumn, Column, ManyToOne, OneToMany } from "typeorm";
import { Thread } from "./Thread";
import { User } from "./User";

@Entity()
export class Post {
    @PrimaryGeneratedColumn()
    id: number;

    @ManyToOne(() => Thread, thread => thread.posts)
    thread: Thread;

    @Column()
    content: string;

    @ManyToOne(() => User, user => user.posts)
    poster: User;

    @Column()
    created: Date;

    @Column()
    updated: Date;

}